modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/PermissionPropagationEntryDLFolderLocalServiceWrapper.java

```diff
@@ -0,0 +1,123 @@
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.internal.service;
+
+import com.liferay.document.library.kernel.model.DLFolder;
+import com.liferay.document.library.kernel.model.DLFolderConstants;
+import com.liferay.document.library.kernel.service.DLFolderLocalServiceWrapper;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.PermissionPropagationEntry;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.security.permission.ResourceActionsUtil;
+import com.liferay.portal.kernel.security.permission.propagator.PermissionPropagator;
+import com.liferay.portal.kernel.service.PermissionPropagationEntryLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceWrapper;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portlet.documentlibrary.constants.DLConstants;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Loc Pham
+ */
+@Component(service = ServiceWrapper.class)
+public class PermissionPropagationEntryDLFolderLocalServiceWrapper
+	extends DLFolderLocalServiceWrapper {
+
+	@Override
+	public DLFolder addFolder(
+			String externalReferenceCode, long userId, long groupId,
+			long repositoryId, boolean mountPoint, long parentFolderId,
+			String name, String description, boolean hidden,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		DLFolder dlFolder = super.addFolder(
+			externalReferenceCode, userId, groupId, repositoryId, mountPoint,
+			parentFolderId, name, description, hidden, serviceContext);
+
+		_inheritPermissionFromParentFolder(dlFolder, serviceContext);
+
+		return dlFolder;
+	}
+
+	private void _inheritPermissionFromParentFolder(
+			DLFolder dlFolder, ServiceContext serviceContext)
+		throws PortalException {
+
+		if (dlFolder.getParentFolderId() ==
+				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+
+			int count =
+				_resourcePermissionLocalService.getResourcePermissionsCount(
+					dlFolder.getCompanyId(), DLConstants.SERVICE_NAME,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(serviceContext.getScopeGroupId()));
+
+			if (count == 0) {
+				return;
+			}
+		}
+
+		String parentClassName = DLFolder.class.getName();
+		long parentFolderId = dlFolder.getParentFolderId();
+
+		if (dlFolder.getParentFolderId() ==
+				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+
+			parentClassName = DLConstants.RESOURCE_NAME;
+			parentFolderId = serviceContext.getScopeGroupId();
+		}
```

Comment:
I think the two `If` blocks could be merged.

Reply:
Fixed it